business_rules/operators.py

```diff
@@ -938,6 +939,63 @@ def is_ordered_by(self, other_value: dict) -> pd.Series:
     def is_not_ordered_by(self, other_value: dict) -> pd.Series:
         return ~self.is_ordered_by(other_value)
 
+    @type_operator(FIELD_DATAFRAME)
+    def target_is_sorted_by(self, other_value: dict) -> pd.Series:
+        """
+         Checking if  SESEQ is  sorted based on SESTDTC within USUBJID
+        """
+        target: str = self.replace_prefix(other_value.get("target"))
+        VAR1: str = self.replace_prefix(other_value.get("user_defined_columns")[0])[0]
+        VAR2: str = self.replace_prefix(other_value.get("user_defined_columns")[1])[0]
+
+        hashMap = {}
+        temp_seseq= []
+        for item in self.value[VAR1]:
+            try:
+                hashMap[item] += 1
+            except KeyError:
+                hashMap[item] = 1
+            finally:
+                temp_seseq.append(hashMap[item])
+
+        if type(item) is str:
+            for i in range(len(temp_seseq)):
+                temp_seseq[i] = chr(ord("`")+temp_seseq[i])
+
+        tracker = copy.copy(self.value[VAR1])
+
+        for i in range(len(self.value[VAR1])):
```

Comment:
If I am understanding correctly, you are sorting by VAR1 then by VAR2 and checking if the values match. There are pandas methods for sorting by multiple values: https://www.geeksforgeeks.org/how-to-sort-a-pandas-dataframe-by-multiple-columns-in-python/ I think that would be a cleaner approach and faster. Additionally it may be better to groupby VAR1 and check the groups are sorted.